docs/getting_started/basics/execute.md

```diff
@@ -1,31 +1,35 @@
-# Executing Your Workflow
+# Workflow Execution
 
-You can execute your workflow in different ways:
+A node in a workflow gets executed when its been computed successfully and outputs a result. Once a node has been executed, it will be **clean** and will not be recomputed unless it gets **dirty**. A node will be dirty whenever a plug value is modified or the workflow gets saved. Shift will use dirty propagation to identify the nodes affected by the modifications done to a node and flag them to be recomputed.
```

Comment:
```suggestion
The execution of node in a Shift workflow relates to the computation of its internal logic and, if performed succesfully, updates its output plugs with the corresponding result. Once the execution of a node completes successfully, it gets flagged as **clean** and will not be recomputed unless it becomes **dirty** again. The dirty status is set whenever a plug value is modified or the workflow gets saved or manually reset via the Shift UI. Shift will use dirty propagation to identify the nodes affected by the modifications done to a node and flag them to be recomputed.
```

Reply:
I changed the description according to what we commented through the chat, please have a look